script_helpers/clone_helpers.py

```diff
@@ -0,0 +1,152 @@
+import os
+import sys
+import shutil
+import yaml
+
+
+METADATA_FILES = {
+    'copy_services': 'copy_services.yml',
+    'following_framework': 'following_framework.yml',
+    'bad_words': 'service_questions_to_scan_for_bad_words.yml'
+}
+
+DATES_FILES = {
+    'application_deadline': 'messages/homepage-sidebar.yml',
+    'clarification_deadline': 'questions/declaration/understandHowToAskQuestions.yml',
+    'go_live': 'questions/declaration/canProvideFromDayOne.yml'
+}
+
+
+def get_fw_name_from_slug(fw_slug):
+    # Hack for G-Cloud which helpfully contains a dash
+    # Hack for Digital Outcomes and Specialists which contains a lower-case 'and'
+    return fw_slug.replace('-', ' ').replace('g cloud', 'G-Cloud').title().replace('And', 'and')
+
+
+def get_nbsp_fw_name_from_slug(fw_slug):
+    # Some places use G-Cloud&bsp;11 for non-breaking spaces
+    # Get the last space and 'replace' it
+    name = get_fw_name_from_slug(fw_slug)
+    last_space_position = name.rindex(" ")
+    return name[:last_space_position] + "&nbsp;" + name[(last_space_position + 1):]
+
+
+class FrameworkContentCloner:
+
+    def __init__(self, framework_family, iteration_number, launch_year):
+        self._launch_year = launch_year
+        self._new_fw_slug = f"{framework_family}-{iteration_number}"
+        self._previous_fw_slug = f"{framework_family}-{iteration_number - 1}"
```

Comment:
Not expecting to have to care about e.g. DOS1 omitting the number altogether, are we?

Reply:
Good point!

Reply:
Have added a commit to handle `iteration_number = 1` as well.